src/structures/ByteArray.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Bolt\structures;
+
+use ArrayAccess, Countable;
+
+/**
+ * Class ByteArray
+ *
+ * @author Michal Stefanak
+ * @link https://github.com/neo4j-php/Bolt
+ * @link https://7687.org/packstream/packstream-specification-1.html#bytes
+ * @package Bolt\structures
+ */
+class ByteArray implements ArrayAccess, Countable
+{
+    private array $bytes = [];
+
+    public function __construct(array $bytes = [])
+    {
+        $this->bytes = $bytes;
+    }
+
+    public function offsetExists($offset): bool
+    {
+        return array_key_exists($offset, $this->bytes);
+    }
+
+    public function offsetGet($offset): ?string
+    {
+        return $this->bytes[$offset] ?? null;
+    }
+
+    public function offsetSet($offset, $value)
+    {
+        if ($offset === null)
+            $this->bytes[] = $value;
+        else
+            $this->bytes[$offset] = $value;
+    }
+
+    public function offsetUnset($offset)
+    {
+        unset($this->bytes[$offset]);
+    }
+
+    public function count(): int
+    {
+        return count($this->bytes);
+    }
+
+    public function __toString(): string
+    {
+        return implode($this->bytes);
+    }
+}
```

Comment:
@transistive What do you think about this class? I'm still not 100% sure about it. Because it's in structures directory but it's not really a structure, it's data type. But what other way do I have to distinguish it's ByteArray. There is already array as list and object as map types. Bytes are usually just a string.

Reply:
Hi @stefanak-michal,

I missed your message. To me, this looks great!

I would just keep the same name as packstream uses: Bytes https://7687.org/packstream/packstream-specification-1.html#bytes

Reply:
thanks